packages/cloud-cognitive/src/components/CreateFullPage/CreateFullPage.js

```diff
@@ -0,0 +1,388 @@
+        const total = createSteps.length;
+        const buttons = [];
+        if (total > 1) {
+          buttons.push({
+            label: backButtonText,
+            onClick: () => setCurrentStep((prev) => prev - 1),
+            kind: 'secondary',
+            disabled: currentStep === 1,
+          });
+          buttons.push({
+            label: cancelButtonText,
+            onClick: () => {
+              setOpen(true);
+            },
+            kind: 'ghost',
+          });
+          buttons.push({
+            label: currentStep < total ? nextButtonText : submitButtonText,
+            onClick: currentStep < total ? handleNext : handleSubmit,
+            disabled: isSubmitDisabled(),
+            kind: 'primary',
+            loading: isSubmitting,
+            className: `${blockClass}__create-button`,
+          });
+          setCreateFullPageActions(buttons);
+        }
+      }
+    }, [
+      getFullPageSteps,
+      children,
+      backButtonText,
+      cancelButtonText,
+      currentStep,
+      onClose,
+      nextButtonText,
+      submitButtonText,
+      onRequestSubmit,
+      isSubmitting,
+      open,
+    ]);
+
+    useEffect(() => {
+      const createSteps = getFullPageSteps();
+      const total = createSteps.length;
+      if (total === 1) {
+        console.warn(
+          `${componentName}: CreateFullPages with one step are not permitted. If you require only one step, please use either the narrow full page, CreateSidePanel, or CreateModal components.`
+        );
+      }
+    }, [getFullPageSteps]);
+
+    const continueToNextStep = () => {
+      setIsSubmitting(false);
+      setCurrentStep((prev) => prev + 1);
+    };
+
+    // returns an array of full page steps
+    const getFullPageSteps = useCallback(() => {
+      const steps = [];
+      const childrenArray = Array.isArray(children) ? children : [children];
+      childrenArray.forEach((child) => {
+        if (isFullPageStep(child)) {
+          steps.push(child);
+        }
+      });
+      return steps;
+    }, [children]);
+
+    // check if child is a full page step component
+    const isFullPageStep = (child) => {
+      if (child && child.props && child.props.type === CREATE_FULL_PAGE_STEP) {
+        return true;
+      }
+      return false;
+    };
+
+    const renderProgressSteps = (childrenElements) => {
+      let childrenArray = Array.isArray(childrenElements)
+        ? childrenElements
+        : [childrenElements];
+      childrenArray = childrenArray.filter((child) => isFullPageStep(child));
+      return (
+        <ProgressIndicator
+          currentIndex={currentStep - 1}
+          spaceEqually
+          vertical
+          className={`${blockClass}__progress-indicator`}>
+          {childrenArray.map((child, stepIndex) => (
+            <ProgressStep
+              label={
+                child.props.title
+                  ? child.props.title
+                  : child.props.children[0].props.title
```

Comment:
We are going to need to add some missing pieces here. It's possible that a child of `child.props.children` is _not_ a `CreateFullPageSection` component. So you're going to need to add a function that checks that it is a true `CreateFullPageSection`. Similar to [here](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/888/files#diff-395705295f2424d1eaa383d0bcadcdbd6834ff1905c85714942a790a1a33cd81R191)

Reply:
Definitely. Right now it's written so that it's either searching for a FullPageStepTitle or a FullPageSectionTitle. Adding the more complex logic wasn't really in scope for this PR... But I definitely do plan on adding and testing...

Reply:
Ok sounds good, what you have now just assumes that either CreateFullPageStep or CreateFullPageSection has a title. And if that isn't the case, things will break.